cli/internal/globby/globby_test.go

```diff
@@ -0,0 +1,520 @@
+package globby
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/spf13/afero"
+)
+
+func setup(files []string) func() {
+	var original = getFileSystem()
+	var fs = afero.NewMemMapFs()
+
+	setFileSystem(fs)
+
+	for _, file := range files {
+		// We don't need the handle, we don't need the error.
+		// We'll know if it errors because the tests will not pass.
+		// nolint:errcheck
+		fs.Create(file)
+	}
+
+	return func() {
+		setFileSystem(original)
+	}
+}
+
+func TestGlobFiles(t *testing.T) {
+	type args struct {
+		basePath        string
+		includePatterns []string
+		excludePatterns []string
+	}
+	tests := []struct {
+		name  string
+		files []string
+		args  args
+		want  []string
+	}{
+		{
```

Comment:
Oh boy! Lots of tests! Does it look like I missed anything obvious? (I wrote tests for lots of bad ideas.)

These tests are intended to specify current behavior and treat that as "correct." There are some oddities that I already believe are simply wrong, which we can choose to change the behavior of in the new implementation.

Reply:
> There are some oddities that I already believe are simply wrong

You're probably right, but this is definitely a good approach so we can be intentional about fixing them.

Reply:
It'll also give us a list of things to look for in configurations so we can throw a warning/help them migrate if we know it's different now.